CodeEdit/Features/Settings/Pages/SearchSettings/SearchSettingsView.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  SearchSettingsView.swift
+//  CodeEdit
+//
+//  Created by Esteban on 12/10/23.
+//
+
+import SwiftUI
+
+struct SearchSettingsView: View {
+    var body: some View {
+        SettingsForm {
+            Section {
+                ExcludedGlobPatternList()
+            } header: {
+                Text("Exclude")
+                Text(
+                    "Add glob patterns to exclude matching files and folders from searches and open quickly. " +
+                    "This will inherit glob patterns from the Exclude from Project setting."
+                )
+            }
+        }
+    }
+}
+
+struct ExcludedGlobPatternList: View {
+    @ObservedObject private var searchSettingsModel: SearchSettingsModel = .shared
+
+    @FocusState private var focusedField: String?
+
+    @State private var selection: GlobPattern?
+
+    func addPattern() {
+        searchSettingsModel.ignoreGlobPatterns.append(GlobPattern(value: ""))
+    }
+
+    func removePattern(_ pattern: GlobPattern?) {
+        let selectedIndex = searchSettingsModel.ignoreGlobPatterns.firstIndex {
+            $0 == selection
+        }
+
+        let removeIndex = searchSettingsModel.ignoreGlobPatterns.firstIndex {
+            $0 == selection
+        }
+
+        searchSettingsModel.ignoreGlobPatterns.removeAll {
+            pattern == $0
+        }
+
+        if selectedIndex == removeIndex && !searchSettingsModel.ignoreGlobPatterns.isEmpty && selectedIndex != nil {
+            selection = searchSettingsModel.ignoreGlobPatterns[
+                selectedIndex == 0 ? 0 : (selectedIndex ?? 1) - 1
+            ]
+        }
+    }
```

Comment:
In SwiftUI, I prefer organizing a struct with properties at the top, followed by the body, and then the methods at the bottom. It adds a nice flow to the code! Like here:
```swift
struct SomeView: View {
    @State private var x: Int = 0

    var body: some View {
        Text(hi())
    }

    func hi() -> String {
        return "hi"
    }
}
```

Reply:
Thanks for sharing!